tracer/src/Datadog.Trace.Manual/Tracer.cs

```diff
@@ -16,14 +17,29 @@ namespace Datadog.Trace
     public sealed class Tracer : ITracer, IDatadogOpenTracingTracer
     {
         private static Tracer? _instance;
+        // we use this object to track when they have changed, so we know whether or not we need to update the values
+        private object? _automaticSettings;
+        private ImmutableTracerSettings? _settings;
 
-        [Instrumented]
+        [Instrumented] // This is only _actually_ instrumented up to 3.6.0 (automatic)
         private Tracer(object? automaticTracer, Dictionary<string, object?> initialValues)
         {
             AutomaticTracer = automaticTracer;
-            Settings = new ImmutableTracerSettings(initialValues);
+
+            // In 3.7.0+ we don't bother populating the dictionary with the settings
+            // seeing as the whole _settings object is going to be thrown away and
+            // repopulated if the customer calls Settings. Kind of annoying...
+            // so to avoid the extra work/allocation we check whether the initial values contains the
+            // agent key, as an easy way of detecting if we're running in this version-conflict mode.
```

Comment:
Why `AgentUriKey`? Is there something special about that settings? Should you include that in this comment?

Reply:
No, there's nothing special about it 😄 It's just one that we already support and probably will _continue_ to support for all future versions